### PR TITLE
feat: replace ya-ts-client with new fetch version

### PR DESCRIPTION
**Pull request by grisha87**
# Scope of changes

- Update the ya-ts-client to a `beta` version
- Since the new beta version utilizes _union types_ instead of enums, a lot of places had to be updated to match the new type requirements
- All _models_ exported by `ya-ts-client` representing the REST API requests/responses payloads are now postfixed with the `DTO` postfix. This aims to help to differentiate a `PaymentApi.AllocationDTO` from `ya-ts-client` from a `golem-js` `Allocation` concept.
- Removed `YagnaMock` from the codebase and replaced it with `ts-mockito` based mocks
- Refactored tests to lower the number of `...Service` tests, focusing on testing the actual components under question (fine-grained unit tests) instead of whole services

# Examples

- A single `hello-world.ts` example was added to test the resulting implementation.

# Remarks

- There are some tests left in `todo`, this is done on purpose to keep the scope of the PR under control

### CODE REVIEW

File: examples/basic/hello-world.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  await agreementPool.run();
+  await market.run(workload, allocation);
+  console.log("Core services started");
+
+  try {
+    const agreement = await agreementPool.getAgreement();
+    payment.acceptPayments(agreement);
+
+    const activity = await Activity.create(agreement, yagna);
+    // To cut down the noise
```
Comment (SewerynKras) [inline examples/basic/hello-world.ts:49]: "Stop listening for new offers"

File: src/network/network.ts [hunk 2]
```diff
@@ -255,6 +253,6 @@ export class Network {
   }
 
   private getUrl() {
-    return this.yagnaApi.net["basePath"];
+    return this.yagnaApi.basePath + "/net-api/v1";
```
Comment (SewerynKras) [inline src/network/network.ts:256]: `this.yagnaApi.net.httpRequest.config.BASE`

File: src/network/service.ts [hunk 3]
```diff
@@ -26,7 +26,8 @@ export class NetworkService {
       const data = await this.yagnaApi.identity.getIdentity();
       networkOwnerId = data.identity;
     }
-    this.network = await Network.create(this.yagnaApi, { ...this.options, networkOwnerId });
+
+    this.network = await Network.create(this.yagnaApi, { ...this.options, networkOwnerId: networkOwnerId });
```
Comment (SewerynKras) [inline src/network/service.ts:30]: why?

File: src/payment/strategy.test.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+describe("SDK provided Payment Filters", () => {
+  describe("acceptAllDebitNotesFilter", () => {
+    it.todo("Accepts all debit notes");
```
Comment (SewerynKras) [inline src/payment/strategy.test.ts:3]: TODO

File: tests/mock/activity.mock.ts [hunk 5]
```diff
@@ -1,18 +1,27 @@
...
+} from "../../src";
 import { ExeScriptRequest } from "../../src/activity/activity";
 import { Readable } from "stream";
-import { YagnaApi } from "../../src/utils";
-import { Agreement } from "../../src/agreement";
 
+/**
+ * @deprecated Use ts-mockito instead of providing a global and central implementation of the activity mock
+ */
 export class ActivityMock extends Activity {
```
Comment (SewerynKras) [inline tests/mock/activity.mock.ts:17]: Remove entirely

File: tests/mock/entities/storage_provider.ts
Comment (SewerynKras) [inline tests/mock/entities/storage_provider.ts:1]: delete and replace with ts-mockito

File: tests/unit/helpers.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * Use it to simulate responses from a "long-polled" API endpoint
+ *
+ * @param response The response to return after "polling time"
+ * @param pollingTimeSec The time to wait before returning the response
+ */
+export const simulateLongPoll = <T>(response: T, pollingTimeSec: number = 1) =>
+  new Promise<T>((resolve) => {
+    setTimeout(() => resolve(response), pollingTimeSec * 1000);
```
Comment (SewerynKras) [inline tests/unit/helpers.ts:41]: Test if this can be reduced to 10ms, why wait a second every time?

File: tests/unit/payment_service.test.ts [hunk 7]
```diff
@@ -17,12 +28,64 @@ const TEST_PAYMENT_TIMEOUT_MS = 1000;
...
+      spentAmount: "0",
+      timestamp: new Date().toISOString(),
+      totalAmount: "10",
+      allocationId: "allocation-id",
+      paymentPlatform: "erc20-holesky-tglm",
+      address: "0xtest",
+    });
+    when(mockYagna.payment).thenReturn(instance(mockPayment));
+
+    /** NOTE: The ID is also used in fixtures, make sure that it matches {@see tests/mock/fixtures/invoices.ts} */
```
Comment (SewerynKras) [inline tests/unit/payment_service.test.ts:53]: Then take the id from the fixture instead of hard-coding it

File: tsconfig.spec.json [hunk 8]
```diff
@@ -19,7 +19,13 @@
     "resolveJsonModule": true
   },
   "exclude": ["dist", "node_modules", "examples", "cypress.config.ts"],
-  "include": ["src/**/*.spec.ts", "tests/unit/**/*.test.ts", "tests/unit/**/*.spec.ts", "tests/jest.d.ts"],
+  "include": [
+    "src/**/*.spec.ts",
+    "tests/unit/**/*.test.ts",
+    "tests/unit/**/*.spec.ts",
+    "tests/jest.d.ts",
+    "node_modules/handlebars/types/index.d.ts"
```
Comment (mgordel) [inline tsconfig.spec.json:27]: ??

Comment (SewerynKras) [inline tsconfig.spec.json:27]: ??

File: tests/mock/activity.mock.ts (hunk 5 above)
Comment (mgordel) [reply tests/mock/activity.mock.ts:17]: i will do

File: tests/mock/index.ts
[the remark this answers is not in the document]
Comment (mgordel) [reply tests/mock/index.ts:1]: I'll do

File: tests/unit/helpers.ts (hunk 6 above)
Comment (SewerynKras) [reply tests/unit/helpers.ts]: @mgordel on my machine changing this long poll to 10ms and logger match timeout to 100ms reduced the test time from 26s->11s

File: src/script/command.ts [hunk 9]
```diff
@@ -1,9 +1,11 @@
-import { ExeScriptRequest } from "ya-ts-client/dist/ya-activity/src/models";
+import { ActivityApi } from "ya-ts-client";
 import { StorageProvider } from "../storage";
-import { Result, ResultState } from "../activity";
+
+// TODO: Check why `import { Result } from "../activity";` does not work correctly here
```
[the remark this answers is not in the document]
Comment (SewerynKras) [reply src/script/command.ts]: @mgordel I did some investigating, and it's most likely some weird deep circular dependency. I could probably untangle it but let's not spend unnecessary time here for 0 benefit

File: tests/unit/helpers.ts (hunk 6 above)
Comment (mgordel) [reply tests/unit/helpers.ts]: good result, let's shorten this time then

File: tsconfig.spec.json (hunk 8 above)
Comment (mgordel) [reply tsconfig.spec.json]: I removed it

**Review by mgordel — APPROVED** [review]

**Pull request merged**